src/main/java/io/confluent/connect/elasticsearch/DataConverter.java

```diff
@@ -120,39 +120,59 @@ static Schema preProcessSchema(Schema schema) {
     }
 
     Schema.Type schemaType = schema.type();
-    Schema keySchema;
-    Schema valueSchema;
     switch (schemaType) {
-      case ARRAY:
-        valueSchema = schema.valueSchema();
-        builder = SchemaBuilder.array(preProcessSchema(valueSchema));
-        return builder.build();
-      case MAP:
-        keySchema = schema.keySchema();
-        valueSchema = schema.valueSchema();
+      case ARRAY: {
+        return copySchemaBasics(schema, SchemaBuilder.array(preProcessSchema(schema.valueSchema()))).build();
+      }
+      case MAP: {
+        Schema keySchema = schema.keySchema();
+        Schema valueSchema = schema.valueSchema();
         String keyName = keySchema.name() == null ? keySchema.type().name() : keySchema.name();
         String valueName = valueSchema.name() == null ? valueSchema.type().name() : valueSchema.name();
-        builder = SchemaBuilder.array(SchemaBuilder.struct().name(keyName + "-" + valueName)
+        Schema elementSchema = SchemaBuilder.struct().name(keyName + "-" + valueName)
             .field(MAP_KEY, preProcessSchema(keySchema))
             .field(MAP_VALUE, preProcessSchema(valueSchema))
-            .build());
-        return builder.build();
-      case STRUCT:
-        builder = SchemaBuilder.struct().name(schema.name());
-        for (Field field: schema.fields()) {
-          builder.field(field.name(), preProcessSchema(field.schema()));
+            .build();
+        return copySchemaBasics(schema, SchemaBuilder.array(elementSchema)).build();
+      }
+      case STRUCT: {
+        SchemaBuilder structBuilder = copySchemaBasics(schema, SchemaBuilder.struct().name(schemaName));
+        for (Field field : schema.fields()) {
+          structBuilder.field(field.name(), preProcessSchema(field.schema()));
         }
-        return builder.build();
-      default:
+        return structBuilder.build();
+      }
+      default: {
         return schema;
+      }
     }
   }
 
+  private static SchemaBuilder copySchemaBasics(Schema source, SchemaBuilder target) {
+    if (source.isOptional()) {
+      target.optional();
+    }
+    if (source.defaultValue() != null && source.type() != Schema.Type.STRUCT) {
```

Comment:
@shikhar To be fair, I'm actually not entirely sure how well this works in practice since you get an "incomplete" schema in the default value. It's definitely not an ideal solution since you can do weird things (e.g. create partial schema, create default value, then add more fields to the schema such that the default value is invalid).